### PR TITLE
remove // from image URLs

### PR DESCRIPTION
**Pull request by struan**
Also cut down on duplicated code by moving the share URL generation code
into a 'private' method.

For mysociety/popit#707

### CODE REVIEW

File: src/mongoose/json-transform.js [hunk 1]
```diff
@@ -78,25 +78,38 @@ function translateDoc(doc, ret, options) {
...
 }
 
+
+function _generateUrl(parts) {
+  var url = parts.join('/');
+
+  // this is a bit grim but there doesn't seem to be a JS
+  // library to do this.
+  var reDoubleSlash = /([^:])\/\//g;
+  url = url.replace(reDoubleSlash, "$1/");
```
Comment (chrismytton) [inline src/mongoose/json-transform.js]: I think you can simplify this by using [`path.join`](http://nodejs.org/api/path.html#path_path_join_path1_path2) which will handle double slashes correctly.

Comment (struan) [inline src/mongoose/json-transform.js]: Sadly this whacks the // in http:// so it's not really helpful.

Comment (chrismytton) [inline src/mongoose/json-transform.js]: Oh :(

**Comment by chrismytton** [comment]
Looks good to me :+1:

**Pull request merged**